x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/pivot/TransformAggregations.java

```diff
@@ -207,14 +208,28 @@ public static Tuple<Map<String, String>, Map<String, String>> getAggregationInpu
         // todo: can this be removed?
         if (agg instanceof PercentilesAggregationBuilder percentilesAgg) {
 
-            // note: eclipse does not like p -> agg.getType()
             // the merge function (p1, p2) -> p1 ignores duplicates
             return new Tuple<>(
                 Collections.emptyMap(),
                 Arrays.stream(percentilesAgg.percentiles())
                     .mapToObj(OutputFieldNameConverter::fromDouble)
+                    .collect(Collectors.toMap(p -> percentilesAgg.getName() + "." + p, p -> percentilesAgg.getType(), (p1, p2) -> p1))
+            );
+        }
+
+        if (agg instanceof RangeAggregationBuilder rangeAgg) {
+
+            // the merge function (p1, p2) -> p1 ignores duplicates
+            return new Tuple<>(
+                Collections.emptyMap(),
+                rangeAgg.ranges()
+                    .stream()
                     .collect(
-                        Collectors.toMap(p -> percentilesAgg.getName() + "." + p, p -> { return percentilesAgg.getType(); }, (p1, p2) -> p1)
+                        Collectors.toMap(
+                            r -> rangeAgg.getName() + "." + AggregationResultUtils.generateKeyForRange(r.getFrom(), r.getTo()),
+                            r -> "range",
+                            (p1, p2) -> p1
+                        )
                     )
             );
```

Comment:
This does not really map the values correctly. Especially if the agg has sub-aggs.

Running locally I continually get `Failed to deduce mapping for...` when there are sub-agg present or not. 

I think how this should be is:

 - `range` by itself should totally map to `long` (not source, as its just providing a count)
 - `range.sub_agg` if it exists should map to the sub-agg type (see ` if (agg.getSubAggregations().size() > 0) ` predicate).

And then we should have some tests that verify how it works with sub aggs or not.

Reply:
> I think how this should be is:
> 
> * `range` by itself should totally map to `long` (not source, as its just providing a count)
> * `range.sub_agg` if it exists should map to the sub-agg type (see `if (agg.getSubAggregations().size() > 0)` predicate).

Done.

> And then we should have some tests that verify how it works with sub aggs or not.

Done both in the unit test and integration test. Please LMK if you see other scenario that would benefit from more testing.